regions/shapes/tests/test_rectangle.py

```diff
@@ -25,6 +25,21 @@ def wcs():
     header = fits.getheader(filename)
     return WCS(header)
 
+def test_corners():
+
+    xc,yc = 2,2
+    angle = 30*u.deg
+    width = 2
+    height = 1
+    reg = RectanglePixelRegion(center=PixCoord(xc, yc),
+                               width=width, height=height, angle=angle)
+
+    y1 = yc + np.cos(angle) * height/2 + np.sin(angle) * width/2
+    x1 = xc + np.cos(angle) * width/2 - np.sin(angle) * height/2
```

Comment:
Personally I don't like tests where the method is re-implemented. It's likely that if you make a mistake with the rotation matrix in the code, you'd make the same mistake re-coding it in the test.

Suggest to just apply the method and assert_allclose on the numbers that come out.

If you can pick a test case where you either know exactly (e.g. 90 deg rotation) or kinda (e.g. 1 deg rotation) what should come out, then put that, but otherwise IMO also just arbitrary output numbers that establish the current behaviour are better than re-implementing the method.

But I know many people disagree and prefer to write tests this way, so no need to argue. Put as you think best and 👍 to merge.

Reply:
I decided I want both kinds of test, so now both exist.  In particular, I wanted at least one case that was not just 90 deg rotations, since the intermediate rotations (and also not 45 deg rotations...) are somewhat nontrivial